src/emulator/gui.ts

```diff
@@ -0,0 +1,50 @@
+import { EmulatorInstance, EmulatorInfo, Emulators } from "./types";
+import * as javaEmulators from "../serve/javaEmulators";
+import { EmulatorRegistry } from "./registry";
+import { DatabaseEmulator } from "./databaseEmulator";
+import { FirestoreEmulator } from "./firestoreEmulator";
+import * as url from "url";
+import { EmulatorHub } from "./hub";
+
+export interface EmulatorGUIOptions {
+  port: number;
+  host: string;
+  hubInfo: EmulatorInfo;
+  projectId: string;
+  auto_download?: boolean;
+}
+
+export class EmulatorGUI implements EmulatorInstance {
+  constructor(private args: EmulatorGUIOptions) {}
+
+  start(): Promise<void> {
+    const { hubInfo, auto_download } = this.args;
```

Comment:
nit: why pull out these two but then reference `this.args.host` and `port` right after?

Reply:
Done.